lib/fs.js

```diff
@@ -735,6 +735,12 @@ fs.write = function(fd, buffer, offset, length, position, callback) {
     if (typeof position === 'function') {
       position = null;
     }
+    if (typeof offset !== 'number') {
+      offset = 0;
+    }
+    if (typeof length !== 'number') {
+      length = buffer.length - offset;
+    }
```

Comment:
Sure thing, fixed in 4062d4cf. I changed both the comment and the documentation to match the implementation.